src/main/java/com/zufar/onlinestore/security/api/SecurityPrincipalProvider.java

```diff
@@ -20,4 +22,8 @@ public UserDto get() {
                 .getPrincipal();
          return userDtoConverter.toDto(userEntity);
```

Comment:
This method fails in the converter because (UserEntity) SecurityContextHolder
                 .getContext()
                 .getAuthentication()
                 .getPrincipal();
returns only the user ID. Therefore, the converter does not convert, but crashes with an error.